src/mongodb/scores.controller.ts

```diff
@@ -2769,17 +2763,27 @@ export class ScoresController {
     return this.scoresService.getAllSessions(id, limit);
   }
 
-  
+
   @ApiExcludeEndpoint(true)
   @Post('/getUsersTargets')
   async GetUsersTargets(@Res() response: FastifyReply, @Body() data: any) {
     try {
-      const {userIds,language}  = data;
-      let recordData = {};
-        for (const userId of userIds) {
-            const userRecord = await this.scoresService.getTargetsByUser(userId, language);
-            recordData[userId] = userRecord ;
-        }
+      const { userIds, language } = data;
+      let recordData = []
+      for (const userId of userIds) {
+        const userRecord = await this.scoresService.getTargetsByUser(userId, language);
+
+        const filteredData = userRecord.map(data => ({
```

Comment:
Please use $project in mongodb query to achieve this rather than filtering this from here using js.